src/ipa_pc/mod.rs

```diff
@@ -701,3 +702,317 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G> for InnerProductArgPC<G,
         Ok(true)
     }
 }
+
+/// Domain etended IPA PC
+#[derive(Derivative)]
+#[derivative(Clone(bound = ""))]
+pub struct DomainExtendedInnerProductArgPC<G: AffineCurve, D: Digest> {
+    _projective: PhantomData<G>,
+    _digest: PhantomData<D>,
+}
+
+impl<G: AffineCurve, D: Digest> PolynomialCommitment<G> for DomainExtendedInnerProductArgPC<G, D> {
```

Comment:
As stated in my comment on ipa_pc/data_structures.rs, the step we are doing here does not depend on the Dlog scheme. It can be done generally, using just linearity of the non-extended scheme. 
However, dlog specifics such as the succinct verifier needed to be implemented.